tests/e2e/system_test/test_mcg_replication_with_disruptions.py

```diff
@@ -227,3 +238,149 @@ def test_replication_with_disruptions(
+        5. Now take backup of Noobaa db using PV backup method
+        6. Remove the log based replication rules, perform some deletion in
+           source bucket. make sure deletion sync doesn't work
+        7. Recover noobaa db from the backup taken in step-5
+        8. Now check if deletion sync works by deleting some objects from
+           source bucket. Note: Expectation is still unclear
+        9. Now patch the bucket to remove complete replication policy and
+           make sure no replication - no deletion sync works
+
+        """
+
+        # entry criteria setup
+        feature_setup_map = setup_mcg_bg_features(
+            num_of_buckets=5,
+            object_amount=5,
+            is_disruptive=True,
+            skip_any_features=["nsfs", "rgw kafka", "caching"],
+        )
+
+        mockup_logger, source_bucket, target_bucket = log_based_replication_setup()
+
+        # upload test objects to the bucket and verify replication
+        upload_test_objects_to_source_and_wait_for_replication(
+            mcg_obj_session,
+            source_bucket,
+            target_bucket,
+            mockup_logger,
+            600,
+        )
+
+        # Delete objects in the first set in a batch and perform noobaa pod
+        # restarts at the same time and make sure deletion sync works
+
+        objs_in_bucket = mockup_logger.standard_test_obj_list
+        objs_to_delete = random.sample(objs_in_bucket, 3)
+
+        from ocs_ci.utility.retry import retry
+
+        @retry(Exception, tries=10, delay=5)
+        def delete_objs_in_batch():
+            for obj in objs_to_delete:
+                mockup_logger.delete_objs_and_log(source_bucket.name, [obj])
+                time.sleep(5)
+            logger.info(f"Successfully deleted these objects: {objs_to_delete}")
+
+        with ThreadPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(delete_objs_in_batch)
+
+            # Restart noobaa pods
+            nb_core_pod = get_noobaa_core_pod()
+            nb_db_pod = get_noobaa_db_pod()
+            nb_core_pod.delete()
+            nb_db_pod.delete()
+            wait_for_pods_to_be_running(pod_names=[nb_core_pod.name, nb_db_pod.name])
+
+            # Wait for the object deletion worker in the BG to completion
+            future.result()
+            assert compare_bucket_object_list(
+                mcg_obj_session,
+                source_bucket.name,
+                target_bucket.name,
+                timeout=600,
+            ), f"Deletion sync failed to complete for the objects {objs_to_delete} deleted in the first bucket set"
+
+        # Take noobaa db backup and remove deletion replication policy for the second bucket set
+        # Get noobaa pods before execution
+        noobaa_pods = get_noobaa_pods()
+
+        # Get noobaa PVC before execution
+        noobaa_pvc_obj = get_pvc_objs(pvc_names=["db-noobaa-db-pg-0"])
```

Comment:
Please consider adding a constant for the name of the noobaa db pvc to reuse in the future